src/gen/fold.rs

```diff
@@ -1544,14 +1544,89 @@ pub fn fold_in_place_kind<V: Folder + ?Sized>(_visitor: &mut V, _i: InPlaceKind)
 }
 # [ cfg ( feature = "full" ) ]
 pub fn fold_item<V: Folder + ?Sized>(_visitor: &mut V, _i: Item) -> Item {
-    Item {
-        attrs: FoldHelper::lift(_i . attrs, |it| { _visitor.fold_attribute(it) }),
-        node: _visitor.fold_item_kind(_i . node),
+    use ::Item::*;
```

Comment:
While the pattern might be nice, turning 3 lines of code into ~100 lines of code isn't that nice.

Reply:
nvm I see below why this is done, its good :)

Reply:
That code is also automatically generated by syn_codegen, so it doesn't really matter how long it is :-).